File: app/companions/app/src/main/java/com/niutao/restaurant/fragment/MenuFragment.java
// MenuFragment.java

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity

class MenuFragment : Fragment() {
    private lateinit var nameEditText: EditText
    private lateinit var descriptionEditText: EditText
    private lateinit var priceEditText: EditText
    private lateinit var imageImageView: ImageView
    private lateinit var addButton: Button

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.menu_fragment, container, false)
        nameEditText = view.findViewById(R.id.name_edit_text)
        descriptionEditText = view.findViewById(R.id.description_edit_text)
        priceEditText = view.findViewById(R.id.price_edit_text)
        imageImageView = view.findViewById(R.id.image_image_view)
        addButton = view.findViewById(R.id.add_button)

        addButton.setOnClickListener {
            val name = nameEditText.text.toString()
            val description = descriptionEditText.text.toString()
            val price = priceEditText.text.toString().toDouble()
            val image = imageImageView.drawable.toString()

            val menu = Menu(name, description, price, image)
            // 上传菜品数据
            Toast.makeText(activity, "菜品添加成功", Toast.LENGTH_SHORT).show()
        }

        return view
    }
}

File: app/companions/app/src/main/java/com/example/nittaro/menu/MenuFragment.kt
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import com.example.nittaro.menu.R

class MenuFragment : Fragment() {
    private lateinit var menuName: TextView
    private lateinit var menuPrice: TextView
    private lateinit var menuDescription: TextView
    private lateinit var menuImage: ImageView
    private lateinit var likeButton: Button

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_menu, container, false)

        menuName = view.findViewById(R.id.menu_name)
        menuPrice = view.findViewById(R.id.menu_price)
        menuDescription = view.findViewById(R.id.menu_description)
        menuImage = view.findViewById(R.id.menu_image)
        likeButton = view.findViewById(R.id.like_button)

        likeButton.setOnClickListener {
            // 点赞逻辑
        }

        return view
    }
}
